app/javascript/packages/document-capture/services/upload.tsx

```diff
@@ -108,7 +103,7 @@ async function upload(payload, { method = 'POST', endpoint, csrf }) {
 
   result.isPending = response.status === 202;
 
-  return /** @type {UploadSuccessResponse} */ (result);
+  return result as UploadSuccessResponse;
```

Comment:
same, this is inferred correctly
```suggestion
  return result;
```